synapse/state/__init__.py

```diff
@@ -365,8 +378,13 @@ def resolve_state_groups_for_events(self, room_id, event_ids):
                 delta_ids=delta_ids,
             ))
 
+        room_version = explicit_room_version
+        if not room_version:
```

Comment:
surely we should never get here for an m.room.create, because we should never have to do state resolution?

In other words: `explicit_room_version` seems redundant.